src/subgraph/index.ts

```diff
@@ -134,4 +143,59 @@ export class Subgraph {
     const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
     return getAllVaults(subgraphEndpoint);
   }
+
+  public async getChainVaultData(): Promise<Vault[]> {
+    const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
+    return getChainVaultData(this.rpcConfig.chainId, subgraphEndpoint);
+  }
+
+  public async getTotalPoolValue() {
+    await this.init();
+    const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
+    const vaults = await getChainVaultData(this.rpcConfig.chainId, subgraphEndpoint);
+    const priceIds = vaults.map((v) => v.depositToken?.pyth?.id);
+    const res = await this.pyth.getLatestPricesFromPyth(priceIds as string[]);
+    const price = res.map((pythPrice) => {
```

Comment:
Can you name this variable like priceInfos or something more meaninful? as `price` is like for one id only, but this is an array.

Reply:
done